packages/docsearch-react/src/DocSearchModal.tsx

```diff
@@ -130,6 +131,23 @@ export function DocSearchModal({
     [favoriteSearches, recentSearches, disableUserPersonalization]
   );
 
+  const sendItemClickEvent = React.useCallback(
+    (item: InternalDocSearchHit) => {
+      if (!state.context.algoliaInsightsPlugin || !item.__autocomplete_id)
+        return;
+
+      const insightsClickParams = buildInsightsClickParams(
+        item,
+        item.__autocomplete_id
+      );
+
+      state.context.algoliaInsightsPlugin.insights.clickedObjectIDsAfterSearch(
+        insightsClickParams
+      );
+    },
+    [state]
```

Comment:
Watching for plugin changes would be more accurate.

```suggestion
    [state.context.algoliaInsightsPlugin]
```

Reply:
Good call